### ISSUE TITLE
(aws-elasticloadbalancingv2): Support for deregistration_delay.connection_termination.enabled property on target group

### ISSUE THREAD

**Issue by r0b0ji** [aws/aws-cdk#24328]
### Describe the feature

The NLB target group has a boolean property `deregistration_delay.connection_termination.enabled` [1]. This is NLB specific flag and not available in ALB [2].

But the current cdk implementation of NetworkTargetGroupProps inherits the deregistrationDelay from BaseTargetGroupProps [3] and BaseTargetGroupProps defines the deregistrationDelay as a Duration type [4]. 

So, it is not possible to set this flag through CDK.

[1] https://docs.aws.amazon.com/elasticloadbalancing/latest/network/load-balancer-target-groups.html
[2] https://docs.aws.amazon.com/elasticloadbalancing/latest/application/load-balancer-target-groups.html
[3] https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-target-group.ts
[4] https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-elasticloadbalancingv2/lib/shared/base-target-group.ts#L39

### Use Case

Aws NLB has this feature to terminate connections to targets where deregistration delay is reached and Aws console recommends to set this flag. Though this flag in available in console, it is not available in CDK.

### Proposed Solution

Move the deregistrationDelay from BaseTargetGroupProps down to NetworkTargetGroupProps and ApplicationTargetGroupProps. In NetworkTargetGroupProps have this a composite typed attribute which has duration for timeout_seconds and also a boolean flag to set connection termination.

### Acknowledgements

- [ ] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.59

### Environment details (OS name and version, etc.)

macos, amazon linux2

**Comment by r0b0ji** [comment]
The workaround is to setAttribute on targetGroup, like

```
targetGroup.setAttribute("deregistration_delay.connection_termination.enabled", "true");
```

**Comment by pahud** [comment]
Yes I think the workaround is pretty straightforward but still good to have this as an option exposed to the surface.
